### DISCUSSION TITLE
customer algorithm

### DISCUSSION THREAD

**Discussion by dakeleblack** [alibaba/GraphScope#2273] [General]
When developing a custom algorithm, the example only shows the use of the built-in message aggregation function. How to customize the custom message aggregation function?

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! And a maintainer will get back to you shortly!
Please feel free to contact us on [DingTalk](https://h5.dingtalk.com/circle/healthCheckin.html?dtaction=os&corpId=dingb4b76e0fa9ccd8d516fd0f0acdf19f55&68e71190=0fe34ec0&cbdbhh=qwertyuiop), WeChat account(graphscope) or [Slack](http://slack.graphscope.io/). We are happy to answer your questions responsively.

**Comment by yecol** [comment]
Hi @bitao666,

We have some built-in aggregation functions for general data types. It is possible to extend your own.
Could you give more details about your scenario or need? 
Or you can contact us via DingTalk.

**Reply by dakeleblack** [reply]
I want to know how to define a custom aggregate function in IncEval function，when compiling it throws exception：C function definition not allowed here，C++ does not support defining functions within functions，how to solve it，thanks！

**Reply by dakeleblack** [reply]
Another question，It seems that the custom algorithm does not use a parallel engine, and currently only executes single-thread， why
